lib/spack/llnl/string.py

```diff
@@ -41,6 +41,20 @@ def comma_and(sequence: List[str]) -> str:
     return comma_list(sequence, "and")
 
 
+def ordinal(number: int) -> str:
+    """Return the ordinal number representation of the provided number.
+
+    Args:
+        number: value to convert to ordinal number
+
+    Returns: ordinal number (representation) of the number
+    """
+    suffixes = {"1": "st", "2": "nd", "3": "rd"}
+    d = str(number)[-1]
+    suffix = suffixes[d] if d in suffixes and number % 100 not in [11, 12, 13] else "th"
+    return f"{number}{suffix}"
```

Comment:
I like to avoid extra `str` and string comparisons if I can:

```suggestion
    ones = number % 10
    tens = number % 100 // 10
    suffixes = ["th", "st", "nd", "rd"]
    suffix = "th" if (tens == 1 or ones > 3) else suffixes[ones]
    return f"{number}{suffix}"
```

or maybe even avoid making a list:

```suggestion
    idx = (number % 10) << 1
    tens = number % 100 // 10
    suffix = "th" if tens == 1 else "thstndrdththththththth"[idx:idx+2]
    return f"{number}{suffix}"
```